### PR TITLE
Tf-idf fix: punctuation removal + lowercasing 

### PR DESCRIPTION
**Pull request by I8dNLo**
Remove punctuation properly + lowercase
Average recall 0.83 -> 0.88

### CODE REVIEW

File: fastembed/sparse/bm25.py [hunk 1]
```diff
@@ -120,11 +124,15 @@ def __init__(
...
 
-        self.punctuation = set(string.punctuation)
+        self.punctuation = set(get_all_punctuation())
         self.stopwords = set(self._load_stopwords(model_dir, self.language))
+
         self.stemmer = get_stemmer(language)
         self.tokenizer = WordTokenizer
 
+        self.num_terms: int = 0
+        self.token_accum: int = 0
```
Comment (joein) [inline fastembed/sparse/bm25.py:134]: redundant?

Comment (I8dNLo) [reply fastembed/sparse/bm25.py:134]: Sure

File: fastembed/common/utils.py [hunk 2]
```diff
@@ -41,3 +42,13 @@ def define_cache_dir(cache_dir: Optional[str] = None) -> Path:
     cache_path.mkdir(parents=True, exist_ok=True)
 
     return cache_path
+
+
+def get_all_punctuation():
```
Comment (joein) [inline fastembed/common/utils.py:47]: type annotation

File: fastembed/common/utils.py [hunk 3]
```diff
@@ -41,3 +42,13 @@ def define_cache_dir(cache_dir: Optional[str] = None) -> Path:
...
+    return set(
+        chr(i) for i in range(sys.maxunicode) if unicodedata.category(chr(i)).startswith("P")
+    )
+
+
+def replace_punctuation(text, punctuation):
```
Comment (joein) [inline fastembed/common/utils.py:53]: type annotation

File: fastembed/sparse/bm25.py [hunk 4]
```diff
@@ -64,6 +66,10 @@
 ]
 
 
+def remove_non_alphanumeric(text):
```
Comment (joein) [inline fastembed/sparse/bm25.py:69]: docstring

File: fastembed/common/utils.py (hunk 3 above)
Comment (joein) [reply fastembed/common/utils.py:53]: defined, but not used?

File: fastembed/sparse/utils/tokenizer.py [hunk 5]
```diff
@@ -4,6 +4,17 @@
 from typing import List
 
 
+class SimpleTokenizer:
+    def __init__(self):
```
Comment (joein) [inline fastembed/sparse/utils/tokenizer.py:8]: redundant method

File: fastembed/sparse/utils/tokenizer.py [hunk 6]
```diff
@@ -4,6 +4,17 @@
...
+        pass
+
+    def tokenize(text: str):
```
Comment (joein) [inline fastembed/sparse/utils/tokenizer.py:11]: return type annotation is missing

File: fastembed/sparse/bm25.py (hunk 4 above)
Comment (joein) [reply fastembed/sparse/bm25.py:69]: maybe it should be in utils as well? 🤔

**Review by joein — CHANGES_REQUESTED** [review]

**Review by joein — APPROVED** [review]

**Pull request merged**